### ISSUE TITLE
floating-nav shows items for non existing sections

### ISSUE THREAD

**Issue by melo0187** [mudassir0909/jsonresume-theme-elegant#49]
When using this theme with my resume.json file I noticed that sections like "Awards" were not rendered, which was what I expected since the awards array in my resume.json file was empty.
What I did not expect was that the floating-nav still showed an entry for the non existing section.

Please don't get me wrong. I do not say this is an error.
If I just remove the awards array entirely form the json file, there is still no awards section, but following that there is also no awards item in the floating-nav. 
This means I could just remove empty arrays from the json file and move along. 

I can not tell whether having an empty awards array in the json file is right or wrong, because `resume test` validates the json file both with an empty awards array and without an awards array.

I had a look at your code and figured out that the interpolate function used by the getFloatingNavItems function in index.js is what causes the floating-nav to show items for non existing sections. I tried messing around with it to make it drop the floating-nav items not only when the array is missing but also in case the array is actually in the json file, but empty. 
However I failed and I have to admit that I do not understand how this interpolate function actually works with the _.reduce function to achieve what it does.
If you could explain to me how your interpolate function in index.js works please?
That would be very kind =)

Thanks in advance for your reply.

**Comment by mudassir0909** [comment]
`interpolate` function returns you the value at a nested path in the object. If the path is invalid, it'll return `undefined`. Let me explain with an example:

```javascript
var batman = {
   organization: {
     name: 'Wayne Enterprises',
     admin: {
       name: 'Bruce Wayne'
     }
   }
};

// Accessing organization's admin's name
console.log(interpolate(batman, 'organization.admin.name')); // same as doing batman.organization.admin.name
//=> Bruce Wayne

// Accessing something that doesn't exist
console.log(interpolate(batman, 'organization.location.address')); 
//=> undefined

/* 
If you did batman.organization.location.address, it'll throw an error
But "interpolate" makes sure to not throw up instead fails silently by returning "undefined"
*/
```

If you understand [_.reduce](http://underscorejs.org/#reduce), you can easily get what's happening in `interpolate` function definition

**Comment by mudassir0909** [comment]
And by the way, Thank you for pointing this out :blush:

My last PR had so many changes, so I overlooked testing the theme properly, will integrate unit tests soon!

**Comment by mudassir0909** [comment]
Fixed as part of d278387764e7d3d1e8b013a2823e776c333e2d42

**Comment by melo0187** [comment]
Thank you very much for taking the time to explain.
